### PR TITLE
-> fixed bug with the iteration and more than 1 vehicle spawning

### LINKED ISSUE #13: morevehicles: Make simulation spawn more vehicles
- vehicles should slow down or stop if the ones before them stop

### LINKED ISSUE #26: ConcurrentModificationException on vehicle leaving a road
Moving vehicles is done like this:
``` java
private boolean moveVehicles(long timeStep) {
    for(DefaultWeightedEdge road : vehicleMap.keySet() ){
        for(IVehicle vehicle : vehicleMap.get(road)){
            vehicle.calculateNextPosition(timeStep, nodeGraphMap, vehicleMap);
        }
    }
```
But in `calculateNextPosition()` this can happen:
``` java
if (tlNode.getColor() == TrafficLightColor.Green) {
    float overLap = nextPossiblePosition - roadLength;
    vehicleMap.get(currentRoad).remove(this);
```
This modifies the `vehicleMap` while it's being iterated, so the thing crashes with an exception when the vehicle tries to leave a road. The design of this has to be reworked so there's no such cyclic dependency (probably the vehicle classes themselves shouldn't do actions to actually move vehicles in the state, this should be done by some kind of VehicleManager).

### PR DESCRIPTION
**Pull request by vezer1993**
fixes #26
closes #13
-> spawning more than 1 vehicle
-> vehicles are aware of each other distances travelled and will move according to the vehicle in front of them
-> random origin node
-> random destination node
-> automatic calculation of shortest path between 2 vertexes in a graph
-> vehicles spawn every N seconds
-> MAX number of vehicles says how many vehicles can be on the road at the time, when the max number is reached only when 1 vehicle leaves a new vehicle can be spawned

### CODE REVIEW

File: src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java [hunk 1]
```diff
@@ -11,15 +11,19 @@
...
     public static void main(String[] args) {
 
         Simulation simulation = new Simulation();
-        MainFrame mainFrame = new MainFrame(worldState);
+
         while(true){
             try{
                 Thread.sleep(TIME_STEP);
                 worldState = simulation.getNextSimulationStep(SIMULATION_TIME_STEP);
+                if(mainFrame == null){
```
Comment (iroq) [inline src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java]: why change it like this?

Comment (vezer1993) [inline src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java]: i was getting an error if I do :
MainFrame.mapContainerPanel.mapPanel.NewStateReceived(worldState);
before the simulation actually creates the worldState object... so I made it like this so it does that only once not each time it enters the while loop (and it's after the first step of simulation so worldstate is not null)

File: src/main/java/thewaypointers/trafficsimulator/simulation/factories/VehicleFactory.java [hunk 2]
```diff
@@ -38,20 +46,40 @@ private RoadEdge getTheFirstRoad(String originNode, HashMap<Node, ArrayList<Road
     }
 
     private String calculateDestinationNode(String originNode) {
-        //TODO: calculate the destination node of the vehicle
-        return "3";
+        List<String> extiNodesAvailable = new ArrayList<>();
```
Comment (iroq) [inline src/main/java/thewaypointers/trafficsimulator/simulation/factories/VehicleFactory.java]: spelling error

Comment (vezer1993) [inline src/main/java/thewaypointers/trafficsimulator/simulation/factories/VehicleFactory.java]: yup, thanks changed that and pushed

**Comment by iroq** [comment]
I'd remove the logging, i think it slows down the simulation a little bit and it's not very helpful right now

File: src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java (hunk 1 above)
Comment (iroq) [inline src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java]: i will change MainFrame so it has a parameterless constructor

**Pull request merged**